### PR TITLE
3.x: constrain upstream requests from take, remove limit operator

### PR DESCRIPTION
**Pull request by davidmoten**
As per discussion in #6569, this PR constrains upstream requests from the `take` operator and removes the `limit` operator.

I've also added a couple of unit tests for coverage of `elementAt` that I didn't include in #6620.

### CODE REVIEW

File: src/main/java/io/reactivex/rxjava3/internal/operators/flowable/FlowableTake.java [hunk 1]
```diff
@@ -103,13 +103,29 @@ public void request(long n) {
             if (!SubscriptionHelper.validate(n)) {
                 return;
             }
-            if (!get() && compareAndSet(false, true)) {
-                if (n >= limit) {
-                    upstream.request(Long.MAX_VALUE);
+            while (true) {
```
Comment (akarnokd) [inline src/main/java/io/reactivex/rxjava3/internal/operators/flowable/FlowableTake.java]: Why not use the limit() internals?

File: src/main/java/io/reactivex/rxjava3/core/Flowable.java [hunk 2]
```diff
@@ -11212,52 +11212,6 @@ public final Completable ignoreElements() {
         return RxJavaPlugins.onAssembly(new FlowableLift<R, T>(this, lifter));
     }
 
-    /**
-     * Limits both the number of upstream items (after which the sequence completes)
-     * and the total downstream request amount requested from the upstream to
-     * possibly prevent the creation of excess items by the upstream.
-     * <p>
```
Comment (akarnokd) [inline src/main/java/io/reactivex/rxjava3/core/Flowable.java:11219]: The Javadoc of take should contain these extra information.

**Comment by akarnokd** [comment]
Why not simply delete `take` and rename `limit`? That way, the tests and documentation of limit would remain intact.

**Comment by davidmoten** [comment]
Indeed! Will do.

**Comment by davidmoten** [comment]
Copied code from `limit` to `take` and fixed javadoc.

**Review by akarnokd — APPROVED** [review]

**Review by vanniktech — APPROVED** [review]

**Pull request merged**